### PR TITLE
Finish docs for GitTree

### PR DESCRIPTION
**Pull request by kshyatt**
Someday @ararslan will demand I stop doccing things...

### CODE REVIEW

**Comment by ararslan** [comment]
> Someday @ararslan will demand I stop doccing things...

Never!

File: base/libgit2/tree.jl [hunk 1]
```diff
@@ -1,11 +1,16 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
 """
-Traverse the entries in a tree and its subtrees in post or pre order.
+    treewalk(f::Function, tree::GitTree, payload=Any[], post::Bool = false)
```
Comment (ararslan) [inline base/libgit2/tree.jl]: Inconsistent spacing around the `=`

File: base/libgit2/tree.jl [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
+
+Traverse the entries in `tree` and its subtrees in post or pre order.
```
Comment (ararslan) [inline base/libgit2/tree.jl]: What does "post or pre order" mean? Might be good to spell that out in layman's terms.

File: base/libgit2/tree.jl [hunk 3]
```diff
@@ -19,21 +24,42 @@ end
...
 function filename(te::GitTreeEntry)
     str = ccall((:git_tree_entry_name, :libgit2), Cstring, (Ptr{Void},), te.ptr)
     str != C_NULL && return unsafe_string(str)
     return nothing
 end
 
+"""
+    filemode(te::GitTreeEntry) -> Cint
+
+Return the UNIX filemode of the object on disk to which `te` refers as an integer.
```
Comment (ararslan) [inline base/libgit2/tree.jl]: What behavior does this have on Windows?

Comment (kshyatt) [reply base/libgit2/tree.jl]: I'm not sure!

Comment (ararslan) [reply base/libgit2/tree.jl]: Do we have a test for it that runs on Windows?

Comment (kshyatt) [reply base/libgit2/tree.jl]: Yeah, I think we do. It's [here](https://github.com/JuliaLang/julia/blob/master/test/libgit2.jl#L877). I think that's still the UNIX filemode?

Comment (ararslan) [reply base/libgit2/tree.jl]: I guess so. I didn't think UNIX filemodes were applicable to Windows, but ¯\\\_(ツ)\_/¯

**Comment by kshyatt** [comment]
Is this good to go?

**Review by ararslan — APPROVED** [review]
